.github/workflows/main.yml

```diff
@@ -16,25 +16,24 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        include:
-          - container: 'ubuntu:22.04'
-            test: 'yes'
-          - container: 'ubuntu:24.04'
-            test: 'yes'
-          - container: 'archlinux:base-devel'
-            test: 'yes'
-          - container: 'fedora:latest'
-            test: 'yes'
-          - container: 'alpine:latest'
-            test: 'no'
-          - container: 'debian:unstable'
-            test: 'yes'
+        container:
+          - 'ubuntu:22.04'
+          - 'ubuntu:24.04'
+          - 'archlinux:base-devel'
+          - 'fedora:latest'
+          - 'alpine:latest'
+          - 'debian:unstable'
+        build:
+          - 'meson'
+          - 'autotools'
```

Comment:
in the previous version we had a flat list, which IMO is cleaner. Can we keep 'test' as is and just add meson/autotools permutations?